Flatiron - Tuples chapter.md

```diff
@@ -0,0 +1,115 @@
+
+**Tuples** are a special kind of tool that combine properties of arrays, dictionaries, and classes but are not interchangeable with any of them.  Yup, you read that right, they're kind of like a Franken-data structure.  
+Like an array tuples allow you to access their contents by index.  
+Like a dictionary tuples allow you to access their contents by name (if you specify names). 
+Like a class or struct tuples have *exactly* the contents described in their declaration and of the type indicated in their declaration.  (This is checked for you by the compiler.  Remember, one of the key aspects of Swift is type-safety).
+
+Arrays and Dictionaries are open-ended: they can contain some elements, thousands of elements or no elements, and the contents can grow or shrink.  They're the right solution when you don't know exactly what you have and need to iterate over the contents to search for an item or do something to each element.
+
+Classes, Structs and Tuples are closed: once they've been declared the compiler will guarantee you exactly what they.  They're the right solution when you want to represent structured data and get help from the compiler to make sure its correct.
+
+##What's a tuple good for?
+Now you're probably wondering why on earth you would want to use a tuple and, if that's where your mind has gone, I can't blame you.  Rest assured that there *are* some circumstances where a tuple is extremely useful, kind of like that weird tool you walk by in the hardware store that twists your mind but, in the right hands and in the right situation, can get the job done like nothing else.
+
+Tuples are the tool of choice when you want to pass highly structured data *to* a function or return highly structured data *from* a function but don't need the additional capabilities of a full-blown class or struct and don't want to go through the trouble of creating one.
+
+The most common time you'll find yourself using a tuple is iterating over a dictionary.  Iterating over an array returns *one* thing (an element) for each pass of the iteration, but iterating over a dictionary returns *two* things for each pass (namely the key and the value) .  Therefore, the iteration returns a tuple to you (with the key and value components correctly typed to how the dictionary was declared).
+
+**iteration**
+```
+// iterate over an array
+for name in list {
+	print(name)
+}
+"Darth"
+"Ben"
+"Luke"
+
+// iterate over a Dictionary (note the tuple)
+for (name, role) in simpleDictionary {
+	print(name + "is the " + role + " in Star Wars")
+}
+Luke is the protagonist in Star Wars
+```
+
+**Returning multiple values from functions**
+You can use a tuple to easily return related data from a function.  For example, here's a function that shows how you can, in a single line of code, return useful related information about a series of numbers that otherwise would take *four* lines of code:
+```
```

Comment:
This is a really great example. The only thing I would add is being more explicit with what it is you're about to do before showing them the code.

I guess you can think of it like a three pronged attack
* Being explicit with what it is you're going to solve with the following code (similar to how I wrote the min, max above).
* Showing them the code (answer to the problem).
* Stepping through the code again being very detailed (as in.. here we are declaring four variables named smallest, largest, average, and median assigned to the value 0.0 which means through type inference their of type `Double`.).